tests/transformer/idx_images_by_name_test.py

```diff
@@ -1,20 +1,36 @@
-"""Test for transforming images by name."""
+"""Tests for transforming the images by name."""
+import filecmp
 import os
 
+from cloudimagedirectory import transformer
 
-def test_list_names_of_images():
+
+def test_list_names_of_images(runner, tmp_path):
     """Run transformer end to end and generate a small list that contains image
     names."""
-
-    assert 0 == os.system(
-        "poetry run cloudimagedirectory-transformer -f "
-        "${PWD}/tests/transformer/testdata/input/raw/azure/eastus.json,"
-        "${PWD}/tests/transformer/testdata/input/raw/google/all.json,"
-        "${PWD}/tests/transformer/testdata/input/raw/aws/af-south-1.json"
-        " -op=${PWD} -dp=${PWD}/tests/transformer/testdata"
-        " --filter.until=none"
+    result = runner.invoke(
+        transformer.run,
+        [
+            "-f",
+            "tests/transformer/testdata/input/raw/google/all.json,tests/transformer/testdata/input/raw/aws/af-south-1.json,tests/transformer/testdata/input/raw/azure/eastus.json",
```

Comment:
I don't know if your linter will allow it, but you should be able to split the line to multiple strings like this:

```
        [
            "-f",
            "tests/transformer/testdata/input/raw/google/all.json,"
            "tests/transformer/testdata/input/raw/aws/af-south-1.json,"
            "tests/transformer/testdata/input/raw/azure/eastus.json",
            ...
    ]
```

The trick is that Python automatically concatenates string literals into one string. For example:

```
>>> x = "aaa" "bbb" "ccc"
>>> print(x)
aaabbbccc
```